### DISCUSSION TITLE
Automigrate not reflecting on prod

### DISCUSSION THREAD

**Discussion by milimyname** [pocketbase/pocketbase#5009] [Q&A]
I followed the automigrate docs and migration along and indeed, I got new migration files but when I deploy them to fly. they don't sync on deployed fly instance. I sshed the pocketbase app and it got new migrations files... I wonder what why it does not get to run migrations? Has anyone faced a similar problem?

```main.go

package main

import (
	"log"
	"os"
	"strings"

	_ "mypackage-pocketbase/migrations"

	"github.com/pocketbase/pocketbase"
	"github.com/pocketbase/pocketbase/plugins/migratecmd"
)

func main() {
	app := pocketbase.New()

	// loosely check if it was executed using "go run"
	isGoRun := strings.HasPrefix(os.Args[0], os.TempDir())

	migratecmd.MustRegister(app, app.RootCmd, migratecmd.Config{
		// enable auto creation of migration files when making collection changes in the Admin UI
		// (the isGoRun check is to enable it only during development)
		Automigrate: isGoRun,
	})

	if err := app.Start(); err != nil {
		log.Fatal(err)
	}
}

```

**Comment by ganigeorgiev** [comment]
>  I sshed the pocketbase app and it got new migrations files

I don't understand what do you mean by this? If you use Go the migration files should be part of the resulting binary, you don't have to upload anything in addition.

Without more details on your deployment setup, what commands you are running and what the migration looks like I'm not sure how to help.

Note that migrations are executed only once. The applied migrations are stored in the `_migrations` table.

**Reply by milimyname** [reply]
You are right. My bad! here is my deployment setup:

```Dockerfile

FROM alpine:latest

ARG PB_VERSION=0.22.12

RUN apk add --no-cache \
    unzip \
    ca-certificates

# download and unzip PocketBase
ADD https://github.com/pocketbase/pocketbase/releases/download/v${PB_VERSION}/pocketbase_${PB_VERSION}_linux_amd64.zip /tmp/pb.zip
RUN unzip /tmp/pb.zip -d /pb/

# uncomment to copy the local pb_migrations dir into the container
COPY ./migrations /pb/pb_migrations 

# COPY ./pb_data /pb/pb_data

# uncomment to copy the local pb_hooks dir into the container
# COPY ./pb_hooks /pb/pb_hooks

ENV CGO_ENABLED=0

EXPOSE 8080

# start PocketBase
CMD ["/pb/pocketbase", "serve", "--http=0.0.0.0:8080"]

```

```fly.yml

name: Fly Deploy
on:
  push:
    branches:
      - main
jobs:
  deploy:
    name: Deploy app
    runs-on: ubuntu-latest
    concurrency: deploy-group # optional: ensure only one action runs at a time
    steps:
      - uses: actions/checkout@v4
      - uses: superfly/flyctl-actions/setup-flyctl@master
      - run: flyctl deploy --remote-only
        env:
          FLY_API_TOKEN: ${{ secrets.FLY_API_TOKEN }}
          
 ```

I wonder if I've been missing a command in my dockerfile...

**Reply by ganigeorgiev** [reply]
Go migrations (and any other Go code in general) can be added only during build.

From the above Dockerfile it seems that you are using the prebuilt executable. In that case migrations can be registered only using the JSVM - https://pocketbase.io/docs/js-migrations/.

**Reply by ganigeorgiev** [reply]
If you want to extend PocketBase with Go then you need to change your Dockerfile to build your custom Go executable. See https://pocketbase.io/docs/go-overview/.
